src/Engine/labyrinthFactory.cpp

```diff
@@ -0,0 +1,31 @@
+//
+// Created by nikita hans on 15/04/2018.
+//
+
+#include "labyrinthFactory.h"
+#include <cstdlib>
+#include <ctime>
+
+
+class LabyrinthFactory {
+
+    int labyrinthFactory::createRandom(labyrinth *labyrinth, int xSize, int ySize) {
+        int x, y;
+        labyrinth = new labyrinth(xSize, ySize);
+        for (x = 0; x < xSize; x++) {
+            for (y = 0; y < ySize; y++) {
+                labyrinth->array[x][y] = getRandomBinaryValue();
+            }
+        }
+        return 0;
+    }
+
+    int labyrinthFactory::getRandomBinaryValue() {
+        srand((unsigned int) time(NULL));
+        double r = ((double) rand() / (RAND_MAX));
+        return int(r);
```

Comment:
Logischer fehler der cast wird immer 0 zurückgeben...

Reply:
hab mich damit nochmal beschäftigt... das Problem ist nicht der cast wie ich zuerst angenommen habe, weil ich dachte rand gibt einen Wert zwischen 0 und RAND_MAX zurück ohne den beiden Grenzen. Diese sind aber mit eingeschlossen. 
http://en.cppreference.com/w/cpp/numeric/random/rand

Das eigentliche Problem ist der Aufruf srand((unsigned int) time(NULL)); in der Schleife. Diese Funktion setzt den Zufallsgenerator immer zurück, sodass du für jeden Aufruf von rand() fast gleiche Werte erhälst. Also man ruft srand((unsigned int) time(NULL)) nur einmal auf, um den Generator mit Anfangswerten zu initialisieren.